### ISSUE TITLE
perf: Set stats for new range in splitTrigger without recomputing using delta

### ISSUE THREAD

**Issue by nvanbenschoten** [cockroachdb/cockroach#4478]
See #4322. This is currently blocked on #4447 and #4454.

This will require threading the current batch's `MVCCStats` into `splitTrigger`, adding this to the 
range's stats, and using these aggregate statistics to compute the delta between the total range's stats and the updated range's stats, which should be the new range's stats. Right now, this would all be possible and we would gain a decent performance boost (estimated 20% on range splits), except that the `SystemBytes` and `SystemCount` fields in the `MVCCStats` would not function correctly in all cases. That said, the two blocking issues, along with a number of other inconsistencies I've dug up while looking into this, show that these two fields are already broken. Because of this, I'm wondering if we should go ahead with the performance improvement while keeping the issue to fix the system fields open.

This optimization could also likely be performed in reverse for the `mergeTrigger`, but this is much lower priority because merges dont happen naturally in CockroachDB.

**Comment by tschottdorf** [comment]
might as well do the same thing for Merge to keep both mechanisms in parity. Otherwise the plumbing would look oddly asymmetrical.

I think it's fine to implement this already. The field is broken now and it's not used for anything, and the fix shouldn't touch your code much. Maybe you can add a test that splits a range, stops the world and then computes both halves of the replicas and their sum.
